### PR TITLE
Implement Zaif Tickers, OrderBook, Trades

### PR DESCRIPTION
**Pull request by tmlee**
- What is the purpose of this Pull Request?
Implement tickers, orderbook, and trades API for Zaif exchange

- What is the related issue for this Pull Request?
https://github.com/coingecko/cryptoexchange/issues/104

- [X] I have added Specs
- [X] (If implementing Market Ticker) I have verified that the `volume` attributes is correct
- [X] (If implementing Market Ticker) I have verified that the `base` and `target` is correct

### CODE REVIEW

File: lib/cryptoexchange/exchanges/zaif/services/trades.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+          "#{Cryptoexchange::Exchanges::Zaif::Market::API_URL}/trades/#{market_pair.base.downcase}_#{market_pair.target.downcase}"
+        end
+
+        def adapt(output, market_pair)
+          output.collect do |trade|
+            tr = Cryptoexchange::Models::Trade.new
+            tr.trade_id  = trade['tid']
+            tr.base      = market_pair.base
+            tr.target    = market_pair.target
+            tr.type      = trade['trade_type'] == 'buy' ? 'bid' : 'sell'
```
Comment (dominicwong617) [inline lib/cryptoexchange/exchanges/zaif/services/trades.rb]: @tmlee another [PR](https://github.com/coingecko/cryptoexchange/pull/168/files) uses `buy` and `sell`, we have to decide whether we want to use `bid` or `buy`.

Comment (tmlee) [reply lib/cryptoexchange/exchanges/zaif/services/trades.rb]: Good one. To keep it consistent is to probably use bid, ask. Any way to abstract this out?

Comment (tmlee) [reply lib/cryptoexchange/exchanges/zaif/services/trades.rb]: @dominicwong617  Any thoughts on maybe solving this in a separate issue?

Comment (tmlee) [reply lib/cryptoexchange/exchanges/zaif/services/trades.rb]: https://github.com/coingecko/cryptoexchange/issues/195

Comment (dominicwong617) [reply lib/cryptoexchange/exchanges/zaif/services/trades.rb]: Let's not merge until we know which one to use.

Comment (dominicwong617) [reply lib/cryptoexchange/exchanges/zaif/services/trades.rb]: https://money.stackexchange.com/questions/1063/can-someone-explain-a-stocks-bid-vs-ask-price-relative-to-current-price

```
Whereas, the bid and ask are the best potential prices that buyers and sellers are willing to transact at: the bid for the buying side, and the ask for the selling side. But, think of the bid and ask prices you see as "tip of the iceberg" prices. That is:
```

I think we should use buy/sell

Comment (Swingcloud) [reply lib/cryptoexchange/exchanges/zaif/services/trades.rb]: other exchanges trades respond we've implemented use `buy/sell`

Comment (tmlee) [reply lib/cryptoexchange/exchanges/zaif/services/trades.rb]: @Swingcloud I will merge this first, issue a fix in a separate PR

**Pull request merged**